Incorrect absolute source map paths on Windows

The logic producing relative request paths in source maps is broken on Windows because it converts backslashes to forward slashes not in all cases. This leads to absolute paths in source maps which prevents debugging from IntelliJ WebStorm (as an example of a problem).

EDIT: There are no real failures. The problem is caused by babel-loader See comments below.
(This manifests as test failures in the ConfigTestCases suite. Example of a failure:)

```
AssertionError: expected Array [
  'dummy:///webpack/bootstrap c6577ccbfbeb94c1e520',
  'dummy:///D:/src/webpack/webpack/test/configCases/filename-template/module-filename-template/test.js',
  'dummy:///D:/src/webpack/webpack/test/configCases/filename-template/module-filename-template/index.js'
] to contain 'dummy:///./test.js'
    at Assertion.fail (D:\src\webpack\webpack\test\js\config\extract-text\issue-14\bundle0.js:1018:18)
    at Assertion.prop.value (D:\src\webpack\webpack\test\js\config\extract-text\issue-14\bundle0.js:903:18)
    at Context.<anonymous> (D:\src\webpack\webpack\test\js\config\filename-template\module-filename-template\bundle0.js:65:22)
```

List of failing test cases:

- ConfigTestCases:filename-template-module-filename-template
- ConfigTestCases:source-map:exclude-chunks-source-map
- ConfigTestCases:source-map:line-to-line
- ConfigTestCases:source-map:sources-array-production
- ConfigTestCases:source-map:sources-array-production-cheap-map

Cause of the problem is code in RequestShortener.js. Pull request will follow.

--- a/lib/RequestShortener.js
+++ b/lib/RequestShortener.js
@@ -5,6 +5,7 @@
 var path = require("path");
 
 function RequestShortener(directory) {
+	directory = directory.replace(/\\/g, "/");
 	var parentDirectory = path.dirname(directory);
 	if(/[\/\\]$/.test(directory)) directory = directory.substr(0, directory.length - 1);
 	if(directory) {
@@ -23,7 +24,7 @@ function RequestShortener(directory) {
 	}
 
 	if(__dirname.length >= 2) {
-		var buildins = path.join(__dirname, "..");
+		var buildins = path.join(__dirname, "..").replace(/\\/g, "/");
 		var buildinsAsModule = currentDirectoryRegExp && currentDirectoryRegExp.test(buildins);
 		var buildinsRegExp = buildins.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
 		buildinsRegExp = new RegExp("^" + buildinsRegExp + "|(!)" + buildinsRegExp, "g");
@@ -40,6 +41,7 @@ module.exports = RequestShortener;
 RequestShortener.prototype.shorten = function(request) {
 	if(!request)
 		return request;
+	request = request.replace(/\\/g, "/");
 	if(this.buildinsAsModule && this.buildinsRegExp)
 		request = request.replace(this.buildinsRegExp, "!(webpack)");
 	if(this.currentDirectoryRegExp)
@@ -48,7 +50,6 @@ RequestShortener.prototype.shorten = function(request) {
 		request = request.replace(this.parentDirectoryRegExp, "!..");
 	if(!this.buildinsAsModule && this.buildinsRegExp)
 		request = request.replace(this.buildinsRegExp, "!(webpack)");
-	request = request.replace(/\\/g, "/");
 	request = request.replace(this.nodeModulesRegExp, "/~/");
 	request = request.replace(this.indexJsRegExp, "$1");
 	return request.replace(/^!|!$/, "");

